PWGLF/Tasks/lambdakzeroanalysis.cxx

```diff
@@ -119,50 +130,102 @@ struct lambdakzeroanalysis {
+              if ((v0.qtarm() < paramArmenterosCut * v0.alpha()) || !boolArmenterosCut) {                     //p82 CERN-THESIS-2014-103 Lambda K0s analysis pp
+                registry.fill(HIST("h3dMassLambda"), 0., v0.pt(), v0.mLambda());               //collision.centV0M() instead of 0. once available
+                registry.fill(HIST("h3dMassAntiLambda"), 0., v0.pt(), v0.mAntiLambda());
+                if (saveDcaHist == 1) {
+                  registry.fill(HIST("h3dMassLambdaDca"), v0.dcaV0daughters(), v0.pt(), v0.mLambda());
+                  registry.fill(HIST("h3dMassAntiLambdaDca"), v0.dcaV0daughters(), v0.pt(), v0.mAntiLambda());
+                }
+              }
+            }
+          }
+        }
+        if (TMath::Abs(v0.yK0Short()) < rapidity) {
+          if (v0.distovertotmom(collision.posX(), collision.posY(), collision.posZ()) * RecoDecay::getMassPDG(kK0Short) < lifetimecut->get("lifetimecutK0S")) {
+            if ((v0.qtarm() > paramArmenterosCut * v0.alpha()) || !boolArmenterosCut) {
+              registry.fill(HIST("h3dMassK0Short"), 0., v0.pt(), v0.mK0Short());
+              if (saveDcaHist == 1) {
+                registry.fill(HIST("h3dMassK0ShortDca"), v0.dcaV0daughters(), v0.pt(), v0.mK0Short());
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  PROCESS_SWITCH(lambdakzeroanalysis, processRun3, "Process Run 3 data", true);
+
+  void processRun2(soa::Join<aod::Collisions, aod::EvSels, aod::CentV0Ms>::iterator const& collision, soa::Filtered<aod::V0Datas> const& fullV0s, MyTracks const& tracks)
   {
     if (!collision.alias()[kINT7]) {
       return;
     }
-    if (!collision.sel7()) {
+    if (eventSelection && !collision.sel7()) {
       return;
     }
+    registry.fill(HIST("hSelectedEventCounter"), 0.5);
 
     for (auto& v0 : fullV0s) {
       //FIXME: could not find out how to filter cosPA and radius variables (dynamic columns)
       if (v0.v0radius() > v0radius && v0.v0cosPA(collision.posX(), collision.posY(), collision.posZ()) > v0cospa) {
         if (TMath::Abs(v0.yLambda()) < rapidity) {
           if (v0.distovertotmom(collision.posX(), collision.posY(), collision.posZ()) * RecoDecay::getMassPDG(kLambda0) < lifetimecut->get("lifetimecutLambda")) {
-            registry.fill(HIST("h3dMassLambda"), collision.centV0M(), v0.pt(), v0.mLambda());
-            registry.fill(HIST("h3dMassAntiLambda"), collision.centV0M(), v0.pt(), v0.mAntiLambda());
-            if (saveDcaHist == 1) {
-              registry.fill(HIST("h3dMassLambdaDca"), v0.dcaV0daughters(), v0.pt(), v0.mLambda());
-              registry.fill(HIST("h3dMassAntiLambdaDca"), v0.dcaV0daughters(), v0.pt(), v0.mAntiLambda());
+            if (TMath::Abs(v0.posTrack_as<MyTracks>().tpcNSigmaStorePr()) < TpcPidNsigmaCut) { //previous 900Gev pp analysis had nSigma< 5 for pt<0.7Gev and tpcNSigmaStorePi<3 for pt>0.7GeV; and no cut on K0S
```

Comment:
No, this is not the way to go, here you should check the `tpcNSigma`, the `Store` value has no real physical meaning unless it is converted to float